.changeset/proud-otters-bow.md

```diff
@@ -0,0 +1,175 @@
+
+  find_additional_modules = true
+  rules = [
+    { type = "ESModule", globs = ["*.js"]}
+  ]
+  ```
+
+  Now, a request to `http://localhost:8787/one.js` would return the contents of `src/one.js`, but a request to `http://localhost:8787/hidden/secret.js` would error with _No such module "hidden/secret.js"._ To include this file, you could expand the `rules` array to be:
+
+  ```toml
+  rules = [
+    { type = "ESModule", globs = ["**/*.js"]}
+  ]
+  ```
+
+  Then, a request to `http://localhost:8787/hidden/secret.js` will return the contents of `src/hidden/secret.js`.
+
+  After this update:
+
+  - Let's put the wrangler.toml back to its original configuration:
+
+  ```toml
+  name = "my-worker"
+  main = "src/index.js
+  ```
+
+  - Now, a request to `http://localhost:8787/one.js` will return the contents of `src/one.js`, but a request to `http://localhost:8787/hidden/secret.js` will ALSO return the contents of `src/hidden/secret.js`. THIS MAY NOT BE WHAT YOU WANT. You can "fix" this in 2 ways:
+
+    1. Remove the inline wildcard import:
+
+    ```js
+    // src/index.js
+    export default {
+    	async fetch() {
+    		const name = new URL(request.url).pathname;
+    		const moduleName = "./" + name;
+    		const value = (await import(moduleName)).default;
+    		return new Response(value);
+    	},
+    };
+    ```
+
+    Now, no extra modules are included in the bundle, and a request to `http://localhost:8787/hidden/secret.js` will throw an error. You can use the `find_additional_modules` feature to include it again.
+
+    2. Don't use the wildcard import pattern:
+
+    ```js
+    // src/index.js
+    import one from "./one.js";
+    import two from "./two.js";
+
+    export default {
+    	async fetch() {
+    		const name = new URL(request.url).pathname;
+    		switch (name) {
+    			case "/one.js":
+    				return new Response(one);
+    			case "/two.js":
+    				return new Response(two);
+    			default:
+    				return new Response("Not found", { status: 404 });
+    		}
+    	},
+    };
+    ```
+
+    Further, there may be some files that aren't modules (js/ts/wasm/text/binary/etc) that are in the folder being included (For example, a `photo.jpg` file). This pattern will now attempt to include them in the bundle, and throw an error. It will look like this:
+
+    `[ERROR] No loader is configured for ".png" files: src/photo.jpg`
+
+    To fix this, simply move the offending file to a different folder.
+
+    In general, we DO NOT recommend using the wildcard import pattern. If done wrong, it can leak files into your bundle that you don't want, or make your worker slightly slower to start. If you must use it (either with a wildcard import pattern or with `find_additional_modules`) you must be diligent to check that your worker is working as expected and that you are not leaking files into your bundle that you don't want. You can configure eslint to disallow dynamic imports like this:
+
+    ```jsonc
+    // .eslintrc.js
```

Comment:
What about generating this file in the c3 templates?

Also:

We have `jsonc` type and JSONc syntax vs `.eslintrc.js`

Maybe we should use flat config?

Reply:
Adressed in https://github.com/cloudflare/workers-sdk/pull/6884/commits/290a077301e217adbdbd2f073c359b9bb31afde0